File: backend/test_upload.py
import requests

# 1. Login to get token
login_url = "http://localhost:8080/api/auth/login"
login_payload = {
    "email": "student@example.com",
    "password": "password"
}
response = requests.post(login_url, json=login_payload)
print("Login Status:", response.status_code)
if response.status_code != 200:
    print("Login Failed:", response.text)
    # try another user
    login_payload["email"] = "aarav.sharma@acronexus.edu"
    response = requests.post(login_url, json=login_payload)
    print("Login Status (aarav):", response.status_code)
    if response.status_code != 200:
        print("Login Failed:", response.text)
        exit(1)

token = response.json().get("data", {}).get("token")
print("Got Token")

# 2. Upload Photo
upload_url = "http://localhost:8080/api/v1/profile/photo"
headers = {
    "Authorization": f"Bearer {token}"
}
files = {
    "file": ("test.jpg", b"dummy image content", "image/jpeg")
}
print("Uploading Photo...")
upload_response = requests.post(upload_url, headers=headers, files=files)
print("Upload Status:", upload_response.status_code)
print("Upload Response:", upload_response.text)

File: backend/test_upload.js
const fs = require('fs');

async function testUpload() {
  try {
    const loginRes = await fetch("http://localhost:8080/api/auth/login", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ email: "student@example.com", password: "password" })
    });
    let loginData = await loginRes.json();
    if (!loginData.success) {
      const loginRes2 = await fetch("http://localhost:8080/api/auth/login", {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ email: "aarav.sharma@acronexus.edu", password: "password" })
      });
      loginData = await loginRes2.json();
    }
    const token = loginData.data?.token;
    if (!token) {
      console.log("Failed to login", loginData);
      return;
    }
    
    // Upload
    const formData = new FormData();
    const blob = new Blob(["test"], { type: "image/jpeg" });
    formData.append("file", blob, "test.jpg");
    
    const uploadRes = await fetch("http://localhost:8080/api/v1/profile/photo", {
      method: "POST",
      headers: {
        "Authorization": `Bearer ${token}`
      },
      body: formData
    });
    
    console.log("Upload status:", uploadRes.status);
    const result = await uploadRes.text();
    console.log("Result:", result);
  } catch (err) {
    console.error(err);
  }
}

testUpload();
